internal/eval/eval.go

```diff
@@ -538,6 +630,25 @@ func evalJSONARRPOP(args []string, store *dstore.Store) []byte {
 	return clientio.Encode(popArr, false)
 }
 
+// insertElementAndUpdateArray add an element at the given index
+// Returns remaining array and error
+func insertElementAndUpdateArray(arr []any, index int, elements []interface{}) (updatedArray []any, err error) {
```

Comment:
What if the array to be inserted is of string type ? As per Redis docs, we need to wrap the string with a single quote as well. Does this method handles that case as well ?